core/server/common/src/main/java/alluxio/master/PrimarySelectorClient.java

```diff
@@ -173,7 +188,7 @@ public void takeLeadership(CuratorFramework client) throws Exception {
       LOG.error(mName + " was interrupted.", e);
       Thread.currentThread().interrupt();
     } finally {
-      mIsLeader.set(false);
+      setState(State.SECONDARY);
```

Comment:
Unrelated to this PR, is there a better way to do the waiting above?

Reply:
hm, the previous logic made it impossible for the leader to step down, leading to a bug if the master loses network connectivity. I've fixed this and also made another necessary change in FaultTolerantAlluxioMasterProcess so that we can successfully transition from primary to secondary.